fix: Improve language and grammar in provisioner weighting doc (#2353)
aws/karpenter: website/content/en/preview/tasks/scheduling.md

## Weighting Provisioners

Karpenter allows you to order your provisioners using the `.spec.weight` field so that the node scheduler will deterministically attempt to schedule with one provisioner before another. Below are a few example use-cases that are now supported with the provisioner weighting semantic.

> ### Reserved Instances
> 
> If you have reserved on-demand instances from EC2 at a discount, you may want to tell Karpenter to prioritize this reserved capacity ahead of other instance types.

Reviewer: ```suggestion
### Savings Plans and Reserved Instances

If you have purchased a [Savings Plan](https://aws.amazon.com/savingsplans/) or [Reserved Instances](https://aws.amazon.com/ec2/pricing/reserved-instances/), you may want to tell Karpenter to prioritize this reserved capacity ahead of other instance types.
```

    values: ["amd64"]
```

### Default Node Configuration

> Pods that contain no node selectors, node affinites, or tolerations can potentially be assigned to any node with any configuration. There may be cases where you require these pods to schedule to a specific capacity type or architecture but assigning the relevant node selectors or affinities to all these workload pods may be too tedious or infeasible. Instead, we want to define a cluster-wide default configuration for nodes launched using Karpenter.

Reviewer: ```suggestion
Pods that do not specify node selectors or affinities can potentially be assigned to any node. There may be cases where you require these pods to schedule to a specific capacity type or architecture but assigning the relevant node selectors or affinities to all these workload pods may be too tedious or infeasible. Instead, we want to define a cluster-wide default configuration for nodes launched using Karpenter.
```

  - key: kubernetes.io/arch
    operator: In
    values: ["arm64"]
  - key: node.kubernetes.io/instance-type
    operator: In
    values: ["c4.large", "c4.xlarge"]
```

{{% alert title="Note" color="primary" %}}

> Based on the way that Karpenter does pod batching and bin packing, it is not guaranteed that Karpenter will always choose the highest priority provisioner given specific requirements. For example, it is possible that a pod that has a node selector on a an `arm64` architecture to be provisioned, causing all other workload pods from that batch to also schedule on that node (even if those pods had no explicit requirement for `arm64` architecture). The behavior may also occur if existing capacity is available, as the kube-scheduler will schedule the pods instead of allowing Karpenter to provision a new node.

Reviewer: ```suggestion
Based on the way that Karpenter performs pod batching and bin packing, it is not guaranteed that Karpenter will always choose the highest priority provisioner given specific requirements. For example, it is possible that a pod that has a node selector on an `arm64` architecture to be provisioned, causing all other workload pods from that batch to also schedule on that node (even if those pods had no explicit requirement for `arm64` architecture). The behavior may also occur if existing capacity is available, as the kube-scheduler will schedule the pods instead of allowing Karpenter to provision a new node.
```

> Based on the way that Karpenter does pod batching and bin packing, it is not guaranteed that Karpenter will always choose the highest priority provisioner given specific requirements. For example, it is possible that a pod that has a node selector on a an `arm64` architecture to be provisioned, causing all other workload pods from that batch to also schedule on that node (even if those pods had no explicit requirement for `arm64` architecture). The behavior may also occur if existing capacity is available, as the kube-scheduler will schedule the pods instead of allowing Karpenter to provision a new node.

Reviewer 2: ```suggestion
Based on the way that Karpenter does pod batching and bin packing, it is not guaranteed that Karpenter will always choose the highest priority provisioner given specific requirements. For example, if a pod can't be scheduled with the highest priority provisioner it will force creation of a node using a lower priority provisioner which may allow other pods from that batch to also schedule on that node. The behavior may also occur if existing capacity is available, as the kube-scheduler will schedule the pods instead of allowing Karpenter to provision a new node.
```

Reviewer 2: No strong feelings here, but this seemed clearer.
